src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/run.py

```diff
@@ -67,25 +71,57 @@ def acr_run(cmd,  # pylint: disable=too-many-locals
         source_location = check_remote_source_code(source_location)
         logger.warning("Sending context to registry: %s...", registry_name)
 
+    yaml_template = None
+    if file == "-":
+        import sys
+        for s in sys.stdin.readlines():
+            yaml_template += s
+        values_content = ""
+
+    if cmd_value:
+        yaml_template = "steps: \n  - cmd: {{ .Values.command }}\n    timeout: {{ .Values.timeout }}\n"
```

Comment:
`az acr run -r MyRegistry --cmd bash /dev/null` doesn't work. Error: `
failed to deserialize task and validate: yaml: unmarshal errors: line 3: cannot unmarshal !!str `None` into int`

Please notice in `acr_run` function, the default `timeout` parameter value is `None`

Reply:
For the command `az acr run --timeout 400 -f - /dev/null` with TTY input
`steps: `
`  - cmd: docker ps`
`    timeout: 500` ,
which timeout takes precedence?